### PR TITLE
Working billing address input on the checkout form

### PR DESCRIPTION
**Pull request by PaitoAnderson**
The goal of this PR is to make it a bit easier for others to use this repo as a template to start their own store, by wiring up the billing address inputs which could be a big job.

- The only input on the checkout form that isn't captured is the newsletter, which could easily be done with an additional checkbox extra field.
- Basic form validation was added using `required` on inputs.
- Customer middle initial input was removed.
- Customer telephone input was added.
- @chec/commerce.js library bumped to latest.

### CODE REVIEW

**Comment by jaepass** [comment]
Thanks Paito! Will give this a test as soon as we can!

File: components/checkout/common/AddressForm.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+              <p className="mb-1 font-size-caption font-color-light">
+                Address line 1*
+              </p>
+              <input
+                required
+                autoComplete="street-address"
+                name={`${type}[street]`}
+                value={street}
+                className="rounded-0 w-100"
+                placeholder="House number, steet, etc."
```
Comment (jaepass) [inline components/checkout/common/AddressForm.js:111]:
```suggestion
                placeholder="House number, steet address, etc."
```

File: components/checkout/common/AddressForm.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+          <div className="col-12 col-sm-6 mb-3">
+            <label className="w-100">
+              <p className="mb-1 font-size-caption font-color-light">
+                Address line 2 (optional)
+              </p>
+              <input
+                name={`${type}[street_2]`}
+                value={street2}
+                className="rounded-0 w-100"
+                placeholder="Appartment, buero, etc."
```
Comment (jaepass) [inline components/checkout/common/AddressForm.js:124]:
```suggestion
                placeholder="Apartment, suite number, etc."
```

File: components/checkout/common/AddressForm.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                    </option>
+                  ))
+                }
+              </Dropdown>
+            </label>
+          </div>
+          <div className="col-12 col-sm-6 mb-3">
+            <label className="w-100">
+              <p className="mb-1 font-size-caption font-color-light">
+                Postal code*
```
Comment (jaepass) [inline components/checkout/common/AddressForm.js:154]:
```suggestion
                Postal code/zip code*
```

File: components/checkout/common/AddressForm.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      townCity,
+      street,
+      street2,
+      postalZipCode,
+    } = this.props;
+
+    return (
+      <>
+        <div className="row">
+          <div className="col-12">
```
Comment (jaepass) [inline components/checkout/common/AddressForm.js:61]:
```suggestion
          <div className="col-12 mb-3">
```

**Review by jaepass — COMMENTED** [review]
Thanks for the PR, works great and much more encapsulated! I've just left some minor suggestions.

**Comment by PaitoAnderson** [comment]
Thanks @jaepass, great suggestions!

**Pull request merged**